### PR TITLE
Laboratory work #3, Badasyan Alexandra - 19FPL1

### PR DESCRIPTION
**Pull request by suralmasha**

### CODE REVIEW

File: lab_3/main.py [hunk 1]
```diff
@@ -17,38 +19,76 @@ def tokenize_by_sentence(text: str) -> tuple:
...
         :return: 0 if succeeds, 1 if not
         """
-        pass
+        if not isinstance(letter, str):
+            return 1
+
+        if letter not in self.storage:
+            letter_id = 0
+            self.storage[letter] = letter_id
+            letter_id += 1
```
Comment (WhiteJaeger) [inline lab_3/main.py]: This `letter_id` is reset to `0` every time this method is called. Find a way to use `letter_id` on the class-level.

File: lab_3/main.py [hunk 2]
```diff
@@ -119,15 +227,41 @@ def _calculate_distance(self, first_n_grams: tuple, second_n_grams: tuple) -> in
...
+        if not isinstance(first_n_grams, tuple) or not isinstance(second_n_grams, tuple):
+            return -1
+        for character_1, character_2 in zip(first_n_grams, second_n_grams):
+            if not isinstance(character_1, tuple) or not isinstance(character_2, tuple):
+                return -1
+
+        distance = 0
+        for index, n_gram in enumerate(first_n_grams):
+            if n_gram in second_n_grams:
+                distance += math.fabs(index - second_n_grams.index(n_gram))
```
Comment (WhiteJaeger) [inline lab_3/main.py]: Why not using python built-in `abs`?

**Review by WhiteJaeger — CHANGES_REQUESTED** [review]

File: lab_3/main.py (hunk 1 above)
Comment (suralmasha) [reply lab_3/main.py]: got it!

File: lab_3/main.py (hunk 2 above)
Comment (suralmasha) [reply lab_3/main.py]: ok

File: lab_3/main.py [hunk 3]
```diff
@@ -17,38 +19,74 @@ def tokenize_by_sentence(text: str) -> tuple:
...
+
+    sentences = re.split(r'\W{1,3}\s(?=[A-Z])', text)
+    result = []
+
+    for sentence in sentences:
+        sent = []
+        for token in re.sub('[^a-z ]', '', sentence.lower()).split():
+            word = []
+            for index, letter in enumerate(token):
+                if index == 0:
```
Comment (WhiteJaeger) [inline lab_3/main.py]: `index == 0` can be shortened.

File: lab_3/main.py [hunk 4]
```diff
@@ -59,49 +97,105 @@ def encode_corpus(storage: LetterStorage, corpus: tuple) -> tuple:
...
-        pass
+        if not self.n_grams:
+            return 1
+
+        all_n_grams = []
+        for text in self.n_grams:
+            for sentence in text:
+                for word in sentence:
+                    all_n_grams.append(word)
+        set_n_grams = set(list(all_n_grams))
```
Comment (WhiteJaeger) [inline lab_3/main.py]: Do you really need to use `set` here?

**Review by WhiteJaeger — CHANGES_REQUESTED** [review]

**Pull request closed without merging**